yearn/contracts/ConvexStrategy.sol

```diff
@@ -0,0 +1,612 @@
+    ///         available by the liquidation.
+    /// @return loss Amount of the loss.
+    function liquidatePosition(uint256 amountNeeded)
+        internal
+        override
+        returns (uint256 liquidatedAmount, uint256 loss)
+    {
+        uint256 wantBalance = balanceOfWant();
+        if (wantBalance < amountNeeded) {
+            liquidatedAmount = withdrawSome(amountNeeded.sub(wantBalance));
+            liquidatedAmount = liquidatedAmount.add(wantBalance);
+            loss = amountNeeded.sub(liquidatedAmount);
+        } else {
+            liquidatedAmount = amountNeeded;
+        }
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to liquidate everything and return the amount that got freed.
+    ///         This strategy implements the aforementioned behavior by withdrawing
+    ///         all vault's underlying tokens from the Convex reward pool.
+    /// @dev This function is used during emergency exit instead of prepareReturn
+    ///      to liquidate all of the strategy's positions back to the vault.
+    /// @return amountFreed Amount that got freed.
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 amountFreed)
+    {
+        // Withdraw all vault's underlying tokens from the Convex reward pool.
+        // However, do not claim the rewards accumulated so far because this is
+        // an emergency action and we just focus on recovering all principle
+        // funds, without trying to realize potential gains.
+        IConvexRewardPool(tbtcConvexRewardPool).withdrawAllAndUnwrap(false);
+
+        // Yearn docs doesn't specify clear enough what exactly should be
+        // returned here. It may be either the total balance after
+        // withdrawAllAndUnwrap or just the amount withdrawn. Currently opting
+        // for the former because of
+        // https://github.com/yearn/yearn-vaults/pull/311#discussion_r625588313.
+        // Also, usage of this result in the harvest method in the BaseStrategy
+        // seems to confirm that.
+        return balanceOfWant();
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to do anything necessary to prepare this strategy for migration,
+    ///         such as transferring any reserve or LP tokens, CDPs, or other
+    ///         tokens or stores of value. This strategy implements the
+    ///         aforementioned behavior by withdrawing all vault's underlying
+    ///         tokens from the Convex reward pool, claiming all rewards
+    ///         accumulated so far, and transferring those rewards to the new
+    ///         strategy.
+    /// @param newStrategy Address of the new strategy meant to replace the
+    ///        current one.
+    function prepareMigration(address newStrategy) internal override {
+        // Just withdraw the vault's underlying token from the Convex reward pool.
```

Comment:
Nope, it is up to date. We have two kinds of tokens here:
- the vault's underlying token, i.e. the Curve pool LP token
- reward tokens distributed by the Convex reward pool, i.e. CRV, CVX, and extra reward token (KEEP) 

The `withdrawAllAndUnwrap(true)` call withdraws the vault's underlying token and claims (`true` parameter does this) all reward tokens. Now, the vault's underlying token should not be transferred manually in this method because this is done by the parent `migrate` method which calls `prepareMigration`. However, all other tokens like our reward tokens must be transferred manually and this is what we're doing here. 

I tried to describe that in this comment but maybe it is still not clear enough. If you have a suggestion on how we can improve it, I'd be happy to do it.

Reply:
Ok, I see now, maybe I would say instead of "There is no need to transfer those tokens to the new strategy" -> "There is no need to transfer underlying tokens to the new strategy... " I thought by "those" it meant rewards tokens as well.